Migrate to rxjs@7.6.0

AmauryBchp opened the pull request.

AmauryBchp committed:
Migrate to rxjs@7.6.0

AmauryBchp committed:
14.0.9

AmauryBchp commented on src/lib/sesame.service.ts:
```diff
@@ -159,14 +159,16 @@ export class SesameService {
         map(([pem, jwt]) => this.checkJwt(jwt, pem)),
-        switchMap(userInfo => this.populateLanguage(userInfo)),
+        switchMap(userInfo => {
+          if (!userInfo.language) {
```
Je veux bien ton avis sur ça @s-petit 
Du fait qu'on a enlevé le IF avec le cookie j'ai rajouté celui-là pour éviter d'appeler la méthode populateLanguage 2 fois au lieu d'une

s-petit replied:
Je ne comprends pas du tout pourquoi vous avez fait ca. Le cookie n'est plus obligatoire ?

Est-ce que ce changement est lié au truc que tu as fait le mois dernier en virant le CookieParam ?

Pour l'instant j'ai du mal a voir ce que ces changements apportent :/

s-petit replied:
Si je comprends le retour de Yoann, c'etait nécessaire pour retrouver la page welcome c'est bien ca ?

s-petit replied:
J'imagine qu'on ne peut pas adapter plutot t-rex ?

yoannwicker replied:
On a essayé mais on n'a pas réussi en tous cas

AmauryBchp replied:
Cette condition if c'est nous qui l'avons ajouté @s-petit quand on a modifié sesame pour la traduction des langues dans Babel

AmauryBchp replied:
Mais en gros, si on essaye de récupérer le user pour faire une action autre raison que le login, le check posait problème car il essayait de récupérer l'info du cookie param sauf qu'il n'avait pas encore été créé

s-petit commented on src/lib/sesame.service.spec.ts:
```diff
@@ -99,7 +99,7 @@ describe('SesameService', () => {
       expect(sesameHttpMock.getPem).toHaveBeenCalledTimes(1);
       expect(sesameHttpMock.getJwtToken).toHaveBeenCalledTimes(0);
-      expect(sesameHttpMock.fetchLanguage).toHaveBeenCalledTimes(1);
+      expect(sesameHttpMock.fetchLanguage).toHaveBeenCalledTimes(0);
```
Pourquoi avoir changé ces tests ? Est-ce que le nouveau comportement est celui qu'on veut ?

Ici, on a pas besoin de recupérer la langue ?

s-petit commented on src/lib/sesame.service.spec.ts:
```diff
@@ -115,7 +115,7 @@ describe('SesameService', () => {
       expect(sesameHttpMock.getJwtToken).toHaveBeenCalledTimes(0);
       expect(sesameHttpMock.fetchLanguage).toHaveBeenCalledTimes(0);
-      expect(sesameHttpMock.check).toHaveBeenCalledTimes(0);
+      expect(sesameHttpMock.check).toHaveBeenCalledTimes(1);
```
Meme remarque pour les 2 autres cas, le comportement change, est-ce qu'on sait pourquoi ?

Cet appel a check est-il nécessaire ?

yoannwicker replied:
Oui l'appel n'était plus fait si on n'est pas authentifié.
Donc on n'obtenait pas d'objet observable UserInfo, et on ne rentrait pas dans le pipe suivant (dans T-Rex) :

```
return this.sesame.userInfo().pipe(
         first(),
         map((userInfo => {
               const isAuthenticated = !!userInfo || !environment.production;
               if (!isAuthenticated) {
                  this.router.navigate(['welcome']);
               }
               return isAuthenticated;
            })
         ));
```

Et donc impossible de rediriger sur la page welcome dans le cas ou on n'est pas authentifié.

s-petit replied:
Ok ca marche je comprends mieux :+1: 

Dans ce cas, je pense qu'il faudrait adapter le titre du test en expliquant clairement comme tu viens de le faire :+1:

mathieupouedras reviewed (approved):
:+1: cool les commit séparés

AmauryBchp committed:
Initialize undefined user info when there is no cookie

Before this fix, the method check was checking if there was a JWT cookie
to construct the zip.

That check created a bug in our backoffice applications (T-rex, Babel,
...) when retrieving user info because we can not get user info if there
is no JWT cookie;

Now, the user info observable is initialized at undefined when there is
no JWT cookie.

fchetouani commented on src/lib/sesame.service.ts:
```diff
@@ -159,6 +159,7 @@ export class SesameService {
   }
 
   private check(): void {
+    this.doOnUserInfo(undefined);
```
:+1: minimaliste :-)

AmauryBchp committed:
14.0.11

yoannwicker reviewed (approved):
:rocket:
